src/env-inl.h

```diff
@@ -101,6 +101,20 @@ inline void Environment::AsyncHooks::set_enable_callbacks(uint32_t flag) {
   fields_[kEnableCallbacks] = flag;
 }
 
+inline Environment::AsyncCallbackScope::AsyncCallbackScope(Environment* env)
+    : env_(env) {
+  env_->async_callback_cntr_++;
+}
+
+inline Environment::AsyncCallbackScope::~AsyncCallbackScope() {
+  env_->async_callback_cntr_--;
+  CHECK_GE(env_->async_callback_cntr_, 0);
+}
+
+inline bool Environment::AsyncCallbackScope::in_tick() {
```

Comment:
@trevnorris Wouldn't any value of `async_callback_cntr` > 0 mean that the current thread is in a `MakeCallback` call? It seems that what we want to check is whether we're in the _first_ (or root)`MakeCallback` call, which is why I had suggested the name `in_nested_makecallback_call` originally.